app/main.py

```diff
@@ -1,3 +1,17 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    results = dict()
+
+    def inner(*args) -> Any:
+        for _ in args:
+            if not type(_) in {int, float, tuple, str}:
+                return inner
```

Comment:
This is not needed. You can ignore this

Reply:
But in task condition it is mentioned that function should get only immutable arguments. No?

Reply:
> Decorator cache creating for decorating only functions that take immutable arguments.
It is means other types can not be send